docs/myria-python/index.md

```diff
@@ -178,6 +178,95 @@ query = MyriaQuery.parallel_import(relation=relation, work=work)
 print query.status
 ```
 
+### Part 4. Python Expressions
+
+Myria allows arbitrary Python Functions as expressions which can be used in transformations like map, flatmap and aggregates.
+
+#### Registering a Python Function
+Python function needs to be registered before it can be used.
+Myria-Python can be used to register new Python functions, list existing registered Python functions,
+and to retrieve details about existing registered functions.
+
+```python
+from myria.udf import MyriaFunction, MyriaPythonFunction
+from raco.types import LONG_TYPE
+
+#create connection
+MyriaRelation.DefaultConnection = MyriaConnection(rest_url='http://demo.myria.cs.washington.edu:8753')
+
+#register Python functions
+@myria_function(output_type="LONG_TYPE")
+def pyIsPrime(dt):
+    import math
+    n = dt[0][0]
+    if n % 2 == 0 and n > 2: 
+        return False
+    for i in range(3, int(math.sqrt(n)) + 1, 2):
+        if n % i == 0:
+            return 0
+    return 1
+
+#List registered functions
+print MyriaFunction.get_all(connection=MyriaRelation.DefaultConnection)
+
+#List details of a registered function
+print MyriaFunction.get('pyIsPrime', connection=MyriaRelation.DefaultConnection)
+
+```
+#### Invoking Python Expression in MyriaL
+A registered Python function can then be used in a MyriaL query anywhere an expression
+ can be used.
+```python
+q = MyriaQuery.submit(""" 
+  T1 = scan(TwitterK);
+  isPrime = [from T1 emit pyIsPrime(T1.src) as isPrime, T1.src, T1.dst];
+  store( isPrime, TwitterK_isPrime);
+""", connection=connection)
+q.status
+
+```
+#### Python Expressions in UDAs
+Python functions can also be used in User Defined Aggregates in MyriaL.
+
+```python
+#register a python function
+@myria_function(output_type="BLOB_TYPE")
+def udfSum(dt):
+    import numpy as np
+    tuplist = dt
+    state = None
+    for i in tuplist:
+        imgid = i[1]
+        subjid = i[0]
+        img = np.asarray(i[2])
+        shape = img.shape + (5,)
+        if state is None:
+            state = np.empty(shape)
+            state[:,:,:,imgid] = img
+        else:
+            state[:,:,:,imgid] = img
+    return (state)
+
+#define a UDA
+
+q = MyriaQuery.submit("""
+  uda aggregate(subjid,imgid, img) {
+    --init
+    [ b'' as tm];
+    --update
+    [udfSum(subjid ,imgid, img )];
+    --emit
+    [tm];
+  };
+  t = scan(public:adhoc:raw);
```

Comment:
This relation doesn't exist on a vanilla Myria install (or on demo) -- need to stage the content on S3 and do an explicit load so that the query works.